### DISCUSSION TITLE
Using Exaile with Traktor Z1

### DISCUSSION THREAD

**Discussion by phalgunlolur** [exaile/exaile#918] [Q&A]
Hello everyone,

I was wondering if it would be possible to use exaile with a Traktor Z1? https://www.native-instruments.com/en/products/traktor/dj-controllers/traktor-kontrol-z1/

I have been looking for a portable 3 band physical EQ (4 or 5 band would be even better) to use with Exaile. So have been considering the Z1, but not sure if I can use it with Exaile. Alternatively, other physical EQ suggestions are welcome

I know virtual DJ is configured to be able to use it as seen here - https://www.virtualdj.com/manuals/hardware/ni/z1/installation.html

**Comment by sjohannes** [comment]
It's not clear to me how the device presents itself to the OS, but if its shows up as a soundcard (two soundcards?) then it will probably work.

Exaile's *previewdevice* plugin lets you use two soundcards to play different tracks, and it is designed for DJing, but I don't know if the workflow is suitable for mixing.

**Comment by phalgunlolur** [comment]
Thanks @sjohannes, I think it works with any software that is capable of receiving MIDI, similar to Xone K2 (the other controller I'm considering) - https://www.thomann.de/gb/allen_heath_xone_k2.htm

So I guess what I'm trying to find out is whether Exaile supports MIDI controls. I'm not an expert on this topic so wouldn't want to risk spending £250 on a product which wont work with Exaile (unless it means that I need to switch softwares)

**Reply by virtuald** [reply]
AFAIK Exaile doesn't support MIDI controls -- but I would expect that only to apply to the various buttons that it has? If you're merely looking for a physical EQ, it's possible the Traktor Z1 or similar devices might work. The key question is whether the EQ requires software to know that it exists to work. If it is just modifying the output of its internal soundcard, then as long as you can tell Exaile to use that soundcard it would work (assuming it presents itself as a normal soundcard on Windows).

You should probably reach out to the manufacturer and ask them.

**Reply by virtuald** [reply]
Quoting from the user manual regarding the EQ knobs:

> These knobs control both TRAKTOR's User-Gain level and the Z1's internal direct gain simultaneously. Note that these knobs only control the gain of Decks A and B.

It sounds like at least some of the features are things that control the internal hardware and don't require the software to work. But really, only the manufacturer or someone with the hardware would be able to answer these questions.
